### PR TITLE
Kyber/Dilithium fips versions -> standard branch

### PR DESCRIPTION
**Pull request by bhess**
PR to merge the Kyber/Dilithium (fips draft versions) to a standard branch. May be obsolete once #1537 is resolved.

* [x] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider), [OQS-OpenSSL](https://github.com/open-quantum-safe/openssl), [OQS-BoringSSL](https://github.com/open-quantum-safe/boringssl), and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: docs/algorithms/sig/dilithium.yml [hunk 1]
```diff
@@ -15,22 +15,16 @@ website: https://pq-crystals.org/dilithium/
 nist-round: 3
 spec-version: 3.1
```
Comment (baentsch) [inline docs/algorithms/sig/dilithium.yml:16]: Looks fishy to me: No change to spec version but different KATs?

File: docs/algorithms/kem/kyber.yml [hunk 2]
```diff
@@ -17,15 +17,9 @@ website: https://pq-crystals.org/
 nist-round: 3
 spec-version: NIST Round 3 submission
```
Comment (baentsch) [inline docs/algorithms/kem/kyber.yml:18]: Looks fishy to me: No change to spec version but different KATs?

**Review by baentsch — CHANGES_REQUESTED** [review]
Can I ask for a change to the version ID of the algs, please? Otherwise https://github.com/open-quantum-safe/oqs-provider/blob/main/oqs-template/oqs-sig-info.md for example will show the same algorithm "Implementation Version" for a different OID: Would be IMO pretty confusing.

File: docs/algorithms/sig/dilithium.yml (hunk 1 above)
Comment (bhess) [reply docs/algorithms/sig/dilithium.yml]: updated in 743c15ceea63ac88631d1ec16bab1ea6061eb6f0

File: docs/algorithms/kem/kyber.yml (hunk 2 above)
Comment (bhess) [reply docs/algorithms/kem/kyber.yml]: updated in 743c15ceea63ac88631d1ec16bab1ea6061eb6f0

**Review by baentsch — APPROVED** [review]
Thanks! LGTM now.

**Comment by baentsch** [comment]
Thanks again for PR and updates -- I'd suggest to wait with merge until Sep 13 to see whether we _really_ have to go this route (or a straight merge as per https://github.com/open-quantum-safe/liboqs/pull/1537#issuecomment-1712127229 is preferred (@dstebila @cryptojedi ?) and we can avoid maintaining separate branches). If we need to do it, I'll then set up a separate branch in `oqsprovider` with the new OIDs you provided to BouncyCastle so you don't have to bother with that too.

**Pull request merged**